cicecore/cicedyn/infrastructure/io/io_netcdf/ice_restart.F90

```diff
@@ -147,7 +147,7 @@ subroutine init_restart_write(filename_spec)
       use ice_dyn_shared, only: kdyn
       use ice_grid, only: grid_ice
 
-      character(len=char_len_long), intent(in), optional :: filename_spec
+      character(len=*), intent(in), optional :: filename_spec
```

Comment:
This is probably only an issue if the filename_spec is longer than "char_len_long".  But agree this is a good change.